### PR TITLE
Add test coverage for ListBox

### PR DESCRIPTION
**Pull request by Syareel-Sukeri**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for ListBox to test its properties: PreferredHeight, Text
- Add unit tests for ListBox to test its method: Refresh, SetSelected

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11624)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 1]
```diff
@@ -6258,6 +6259,205 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+    public void ListBox_GetItemText_FormattingEnabledFalse_ReturnsItemText()
+    {
+        using ListBox listBox = new() { FormattingEnabled = false };
+        listBox.Items.Add("TestItem");
+        listBox.SelectedItem = "TestItem";
+
+        string result = listBox.GetItemText(listBox.SelectedItem);
+        result.Should().Be("TestItem");
+    }
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6292]: Maybe that we can merged the ListBox_GetItemText_FormattingEnabledTrue_SelectedItemNull_ReturnsEmptyString() & ListBox_GetItemText_FormattingEnabledTrue_SelectedItemNotNull_ReturnsItemText() & ListBox_GetItemText_FormattingEnabledFalse_ReturnsItemText() into one test case.

    [WinFormsTheory]
    [InlineData(true, null, "")]
    [InlineData(true, "TestItem", "TestItem")]
    [InlineData(false, "TestItem", "TestItem")]
    public void ListBox_GetItemText_ReturnsExpected(bool formattingEnabled, string selectedItem, string expected)
    {
        using ListBox listBox = new() { FormattingEnabled = formattingEnabled };
        if (selectedItem is not null)
        {
            listBox.Items.Add(selectedItem);
            listBox.SelectedItem = selectedItem;
        }

        string result = listBox.GetItemText(listBox.SelectedItem);
        result.Should().Be(expected);
    }

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 2]
```diff
@@ -6258,6 +6259,205 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+        listBox.MeasureItem += (sender, e) =>
+        {
+            measureItemCalled = true;
+        };
+
+        listBox.Refresh();
+
+        measureItemCalled.Should().BeFalse();
+    }
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6352]: Please merged the cases ListBox_Refresh_OwnerDrawVariable_CallsOnMeasureItem() &  ListBox_Refresh_OwnerDrawFixed_DoesNotCallOnMeasureItem() & ListBox_Refresh_Normal_DoesNotCallOnMeasureItem() into one.

    [WinFormsTheory]
    [InlineData(DrawMode.OwnerDrawVariable, true)]
    [InlineData(DrawMode.OwnerDrawFixed, false)]
    [InlineData(DrawMode.Normal, false)]
    public void ListBox_Refresh_CallsOnMeasureItemAsExpected(DrawMode drawMode, bool expectedMeasureItemCalled)
    {
        using ListBox listBox = new()
        {
            DrawMode = drawMode,
            Items = { "Item1", "Item2", "Item3" }
        };

        bool measureItemCalled = false;
        listBox.MeasureItem += (sender, e) =>
        {
            measureItemCalled = true;
        };

        listBox.Refresh();

        measureItemCalled.Should().Be(expectedMeasureItemCalled);
    }

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 3]
```diff
@@ -6258,6 +6259,205 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+    [WinFormsFact]
+    public void ListBox_ItemsCollectionIsNull_ReturnsZero()
+    {
+        using ListBox listBox = new();
+        listBox.Items.Clear();
+        int itemCount = (listBox.Items is null) ? 0 : listBox.Items.Count;
+        itemCount.Should().Be(0);
+    }
+
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6361]: This is case have the same test content with ListBox_ItemsCollectionIsEmpty_ReturnsZero(), please have a check.

And also please merged the case and ListBox_ItemsCollectionIsNotNull_ReturnsItemCount() into one.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 4]
```diff
@@ -6258,6 +6259,205 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+    public void ListBox_PreferredHeight_RecreatingHandle_ReturnsCurrentHeight()
+    {
+        using ListBox listBox = new()
+        {
+            DrawMode = DrawMode.Normal,
+            BorderStyle = BorderStyle.None,
+            Items = { "Item 1", "Item 2", "Item 3" }
+        };
+
+        var recreateHandleMethod = typeof(ListBox).GetMethod("RecreateHandle", BindingFlags.NonPublic | BindingFlags.Instance);
```
Comment (Zheng-Li01) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6421]: Please using TestAccessor() to access the private/internal method. https://github.com/dotnet/winforms/pull/11361#discussion_r1597878531

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 5]
```diff
@@ -6258,6 +6258,144 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+        measureItemCalled.Should().Be(expectedMeasureItemCalled);
+    }
+
+    [WinFormsTheory]
+    [InlineData(new string[] { }, 0)]
+    [InlineData(new string[] { "Item1", "Item2" }, 2)]
+    public void ListBox_ItemsCollection_ReturnsExpectedCount(string[] items, int expectedCount)
+    {
+        using ListBox listBox = new();
+        listBox.Items.Clear();
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6311]: This line is unnecessary

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs [hunk 6]
```diff
@@ -6258,6 +6258,143 @@ public void ListBox_Remove_SelectedItems(SelectionMode mode, bool createControl)
...
+    {
+        using ListBox listBox = new()
+        {
+            DrawMode = DrawMode.Normal,
+            BorderStyle = BorderStyle.None,
+            Items = { "Item 1", "Item 2", "Item 3" }
+        };
+
+        var accessor = listBox.TestAccessor();
+        accessor.Dynamic.SetState(1, true);
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ListBoxTests.cs:6385]: Please add a comment for this code line

**Review by lonitra — APPROVED** [review]
Thank you

**Pull request merged**